src/main/java/MainWindow.java

```diff
@@ -28,9 +36,29 @@ public void initialize() {
         scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
     }
 
-    /** Injects the Duke instance */
+    /**
+     * Injects the Duke instance.
+     * This method also sets up the UI, parser, task list, and storage for the application.
+     * It also attempts to load any existing data from a file and displays a welcome message.
+     *
+     * @param b The Bro instance to be set for the application.
+     */
     public void setBro(Bro b) {
         bro = b;
+        ui = new Ui();
+        parser = new Parser(ui);
+        tasks = new TaskList(ui, parser);
+        storage = new Storage("./src/main/java/data.txt", tasks);
```

Comment:
I noticed that the file path for data.txt is hardcoded in storage = new Storage("./src/main/java/data.txt", tasks);. Would it be helpful to extract this path into a constant or configuration file so that it can be easily modified without changing the code?